samples/web/content/apprtc/js/roomselection_test.js

```diff
@@ -0,0 +1,114 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+/* More information about these options at jshint.com/docs/options */
+
+/* globals UI_CONSTANTS, RoomSelection, assertMatch, assertTrue, assertEquals,
+   TestCase */
+
+'use strict';
+
+var RoomSelectionTest = new TestCase('RoomSelectionTest');
+
+RoomSelectionTest.prototype.setUp = function() {
+  var key = 'testRecentRoomsKey';
+  localStorage.removeItem(key);
+  localStorage.setItem(key, '["room1", "room2", "room3"]');
+  
+  var targetDiv = document.createElement('div');
+  targetDiv.id = UI_CONSTANTS.roomSelectionDiv.substring(1);
+  
+  var inputBox = document.createElement('input');
+  inputBox.id = UI_CONSTANTS.roomSelectionInput.substring(1);
+  inputBox.type = 'text';
+  
+  var randomButton = document.createElement('button');
+  randomButton.id = UI_CONSTANTS.roomSelectionRandomButton.substring(1);
+  
+  var joinButton = document.createElement('button');
+  joinButton.id = UI_CONSTANTS.roomSelectionJoinButton.substring(1);
+  
+  var recentList = document.createElement('ul');
+  recentList.id = UI_CONSTANTS.roomSelectionRecentList.substring(1);
+  
+  targetDiv.appendChild(inputBox);
+  targetDiv.appendChild(randomButton);
+  targetDiv.appendChild(joinButton);
+  targetDiv.appendChild(recentList);
+  
+  this.roomSelection_ = new RoomSelection(targetDiv, UI_CONSTANTS, key);
+};
+
+RoomSelectionTest.prototype.tearDown = function() {
+  this.roomSelection_ = null;
+};
+
+RoomSelectionTest.prototype.testInputFilter = function() {
+  var validInputs = [ '123123', 'asdfs3', 'room1', '3254234523452345234523452345asdfasfdasdf'];
+  var invalidInputs = ['', ' ', 'abcd', '123', '[5afasdf', 'ñsaer3'];
+  
+  var testInput = function(roomSelection, input, expectedResult) {
+    roomSelection.roomIdInput_.value = input;
```

Comment:
I'd avoid depending on private members of the tested class, since it makes the tests harder to maintain.

Since the test class knows all the DOM elements, can it just interact with them directly and test the public interface of RoomSelection?